Lib/concurrent/futures/process.py

```diff
@@ -306,9 +299,7 @@ def weakref_cb(_, thread_wakeup=self.thread_wakeup):
         #     {5: <_WorkItem...>, 6: <_WorkItem...>, ...}
         self.pending_work_items = executor._pending_work_items
 
-        # Set this thread to be daemonized
         super().__init__()
-        self.daemon = True
```

Comment:
I'm actually depending on this behavior and my code breaks in Python 3.9. Is there a way to set the threads to be daemon? The error I got:

```
Traceback (most recent call last):
  File "/Users/laike9m/.pyenv/versions/3.9.0/lib/python3.9/threading.py", line 950, in _bootstrap_inner
    self.run()
  File "/Users/laike9m/.pyenv/versions/3.9.0/lib/python3.9/threading.py", line 888, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/laike9m/.pyenv/versions/cyberbrain-dev@3.9.0/lib/python3.9/site-packages/grpc/_server.py", line 875, in _serve
    if not _process_event_and_continue(state, event):
  File "/Users/laike9m/.pyenv/versions/cyberbrain-dev@3.9.0/lib/python3.9/site-packages/grpc/_server.py", line 839, in _process_event_and_continue
    rpc_state, rpc_future = _handle_call(event, state.generic_handlers,
  File "/Users/laike9m/.pyenv/versions/cyberbrain-dev@3.9.0/lib/python3.9/site-packages/grpc/_server.py", line 749, in _handle_call
    return _handle_with_method_handler(rpc_event, method_handler,
  File "/Users/laike9m/.pyenv/versions/cyberbrain-dev@3.9.0/lib/python3.9/site-packages/grpc/_server.py", line 722, in _handle_with_method_handler
    return state, _handle_unary_stream(rpc_event, state,
  File "/Users/laike9m/.pyenv/versions/cyberbrain-dev@3.9.0/lib/python3.9/site-packages/grpc/_server.py", line 640, in _handle_unary_stream
    return thread_pool.submit(_stream_response_in_pool, rpc_event, state,
  File "/Users/laike9m/.pyenv/versions/3.9.0/lib/python3.9/concurrent/futures/thread.py", line 163, in submit
    raise RuntimeError('cannot schedule new futures after '
RuntimeError: cannot schedule new futures after interpreter shutdown
```

which I believe is relevant. @aeros

Reply:
@laike9m Hmm, does your code specifically rely on submitting new futures *after* `executor.shutdown()` gets called? If so, would it be possible to move the `executor.shutdown()` to a later point (optimally, you should be done with using the executor before calling `shutdown()`)?

That being said, there may be some reason to consider allowing users to opt-in to using daemon threads; particularly since the decision to prevent subinterpreters from using daemon threads was reverted (which was originally the main motivation for this PR, with the benefit of more predictable shutdown behavior).

Do you have any thoughts @pitrou?

Reply:
@laike9m What do you mean with "break"? Does the process not exit cleanly?

Reply:
Let me describe my use case.

I'm using gRPC which starts a server using `ThreadPoolExecutor`:
https://github.com/laike9m/Cyberbrain/blob/master/cyberbrain/rpc_server.py#L235

This server runs in another thread besides the main thread, and would wait using [`server.wait_for_termination()`](https://github.com/laike9m/Cyberbrain/blob/master/cyberbrain/rpc_server.py#L256) until users explicitly ctrl-c. The main thread would run and finish the work, clients may or may not connect to the server before the main thread end, and the requirement is for the server to respond even after the main thread has finished. In Python <=3.8 this works, but in 3.9 I got the above error.

I'm not 100% that the root cause is this change, but I scanned through all changes in concurrent.futures in 3.9, and this seems to be the most likely one.

Reply:
I see, then I think you need to define a signal handler that will be called when Ctrl-C is pressed, and that will ask the server to stop gracefully.

Reply:
emm, it's actually not about what happens when users press Ctrl+C, but the server is not able to serve requests before that...

```
server starts listening --> main thread finishes ---> requests come ------> Ctrl + C
                                                    (problem is here)
```

Reply:
I'll try to come up with a minimal example that can reproduce this error.

Reply:
Ah, then it's simpler: you should ask the server to finish _and_ wait for it to finish before you let the main thread exit.

Reply:
Sounds good, thanks.